app.coffee

```diff
@@ -54,14 +55,40 @@ app.post "/webhook", (req, res) ->
   # Send a blank response, so slack knows we got it.
   res.send ""
 
+# Ban the nick
+app.post "/user/nick/ban", (req, res)->
+  throw "Invalid Token" unless req.body.token == process.env.OUTGOING_TOKEN_BOT
+
+  if user.ban_nick req.body.nick
+    res.send "ok"
+    return
+
+  res.send "error!"
+
+# Ban the session
+app.post "/user/session/ban", (req, res)->
+  throw "Invalid Token" unless req.body.token == process.env.OUTGOING_TOKEN_BOT
+
+  if user.ban_session req.body.nick
+    res.send "ok"
+    return
+
+  res.send "error!"
```

Comment:
What is this? All slack messages (that reach jinora via slack) are already authenticated, iirc. There is no need to create a new endpoint for ban. Just have a keyword match for slack messaages instead:

`jinora ban admin`

And the user would be banned.